share/spice/what3words/what3words.js

```diff
@@ -1,14 +1,24 @@
 (function (env) {
     "use strict";
     env.ddg_spice_what3words = function (api_result) {
-
-        if (!api_result || !api_result.geometry) {
+        if (!api_result) {
             return Spice.failed('what3words');
         }
+        else if (api_result.status.code == 300 && api_result.status.message === "Invalid or non-existent 3 word address"){
+            // Grab 3 word address from query
+            var query = DDG.get_query()
+            console.log(query)
+            var three_word_re = /[a-z][a-z]+\.[a-z]+\.[a-z]+/;
+            var address = query.match(three_word_re)[0]
+        }
+        else {
+            var success = 'yes'
```

Comment:
@vovkman this variable is only defined conditionally. Meaning "success" on line 18 has the potential to be undefined which we don't want.

You want to initialize the variable, then set it conditionally.